lib/model.ts

```diff
@@ -47,21 +46,19 @@ export class Model extends GenType {
       }
     }
 
-    this.isObject = type === 'object' || !!schema.properties || (schema.allOf || []).length > 0;
+    this.isObject = (type === 'object' || !!schema.properties) && !schema.nullable;
```

Comment:
`nullable` here bit concerns me, `nullable` is unrelated to whether the schema describes an object.

And `nullable` combined with `anyOf`/`oneOf` is implied to be illegal, but many implementations support it.  https://github.com/pglass/py-openapi-schema-to-json-schema/issues/10 has some details about this.